pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -39,3 +44,266 @@ type StorageVolumeList struct {
+
+	// Deduplication to be supported or not
+	// +nullable
+	Deduplication *bool `json:"deduplication,omitempty"`
+}
+
+type StorageVolumeAccessMode string
+
+const (
+	// can be mounted in read/write mode to exactly 1 host
+	ReadWriteOnce StorageVolumeAccessMode = "ReadWriteOnce"
+	// can be mounted in read-only mode to many hosts
+	ReadOnlyMany StorageVolumeAccessMode = "ReadOnlyMany"
+	// can be mounted in read/write mode to many hosts
+	ReadWriteMany StorageVolumeAccessMode = "ReadWriteMany"
+)
+
+
+type RecoveryTimeObjectives string
+
+const (
+	// OnlineActive indicates that the recovery time objective for a volume is online active
+	OnlineActive RecoveryTimeObjectives = "OnlineActive"
+
+	// OnlinePassive indicates that the recovery time objective for a volume is online active
+	OnlinePassive RecoveryTimeObjectives = "OnlinePassive"
+)
+
+type ProvisioningPolicy string
+
+const (
+	// ProvisioningPolicyThick indicates that the provisioning policy is thick
+	ProvisioningPolicyThick ProvisioningPolicy = "Thick"
+
+	// ProvisioningPolicyThin indicates that the provisioning policy is thin
+	ProvisioningPolicyThin ProvisioningPolicy = "Thin"
+)
+
+type CompressionAlgorithm string
+
+const(
+	HuffmanCompression CompressionAlgorithm = "huffman"
+)
+
+type DataSecurityCapabilities struct {
+	MediaEncryption *bool `json:"mediaEncryption,omitempty"`
+}
+
+type IOConnectivityCapabilities struct {
+	AccessProtocols string `json:"accessProtocols,omitempty"`
+}
+
+type IOPerformanceCapabilities struct {
+	AverageIOOperationLatencyMicroseconds string `json:"averageIOOperationLatencyMicroseconds,omitempty"`
+}
+
+type DataProtectionCapabilities struct {
+	// TODO: Decide the Fields
+}
+
+// Affinity is a group of affinity scheduling rules
+type Affinity struct {
+
+	// Describes volume affinity scheduling rules
+	// +optional
+	VolumeAffinity []VolumeAffinityTerm `json:"volumeAffinity,omitempty"`
+
+	// Describes volume anti-affinity scheduling rules
+	// +optional
+	VolumeAntiAffinity []VolumeAffinityTerm `json:"volumeAntiAffinity,omitempty"`
+
+	// Describes cohort affinity scheduling rules
+	// +optional
+	CohortAffinity []metav1.LabelSelector `json:"cohortAffinity,omitempty"`
+
+	// Describes cohort anti-affinity scheduling rules
+	// +optional
+	CohortAntiAffinity []metav1.LabelSelector `json:"cohortAntiAffinity,omitempty"`
+
+	// TopologySpreadConstraint specifies how to spread matching volumes among the given topology.
+	// +optional
+	TopologySpreadConstraint []corev1.TopologySpreadConstraint `json:"topologySpreadConstraint,omitempty"`
+}
+
+type VolumeAffinityTerm struct {
+	TopologyKey string `json:"topologyKey,omitempty"`
+
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+}
+
+// StorageVolumeStatus defines the observed state of StorageVolume
+// +k8s:openapi-gen=true
+type StorageVolumeStatus struct {
+	// Phase defines the state of the volume
+	Phase StorageVolumePhase `json:"phase"`
```

Comment:
Do we need conditions as well in addition to phase?

Reply:
JFYI: pattern of using Phase is deprecated, using conditions is encouraged.   https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties